packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -66,11 +65,11 @@ export class ResourceDownloadQueue {
         } else {
           message = await this.downloadAllPreviewUrls(message, operation);
         }
-
         this.isActive = false;
         this._context.setChatMessage(threadId, message);
       } catch (error) {
         console.log('Downloading Resource error: ', error);
+        this.isActive = false;
```

Comment:
Could this line be placed in a `finally` block?

https://www.w3schools.com/jsref/jsref_try_catch.asp